Common/SymbolRepresentation.cs

```diff
@@ -338,40 +339,39 @@ public static Symbol ParseOptionTickerOSI(string ticker, SecurityType securityTy
         /// <returns>Symbol object for the specified OSI option ticker string</returns>
         public static Symbol ParseOptionTickerOSI(string ticker, SecurityType securityType, OptionStyle optionStyle, string market)
         {
-            var optionTicker = ticker.Substring(0, 6).Trim();
-            var expiration = DateTime.ParseExact(ticker.Substring(6, 6), DateFormat.SixCharacter, null);
-            OptionRight right;
-            if (ticker[12] == 'C' || ticker[12] == 'c')
-            {
-                right = OptionRight.Call;
-            }
-            else if (ticker[12] == 'P' || ticker[12] == 'p')
-            {
-                right = OptionRight.Put;
-            }
-            else
+            
+            var match = Regex.Match(ticker, @"^([A-Z]+)\s*(\d{6})([CP])(\d{8})$", RegexOptions.IgnoreCase);
+            if (!match.Success)
             {
-                throw new FormatException(Messages.SymbolRepresentation.UnexpectedOptionRightFormatForParseOptionTickerOSI(ticker));
+                throw new FormatException("Invalid ticker format.");
             }
-            var strike = Parse.Decimal(ticker.Substring(13, 8)) / 1000m;
+
+            var optionTicker = match.Groups[1].Value;
+            var expiration = DateTime.ParseExact(match.Groups[2].Value, "yyMMdd", null);
+            OptionRight right = match.Groups[3].Value.ToUpper() == "C" ? OptionRight.Call : OptionRight.Put;
+            var strike = decimal.Parse(match.Groups[4].Value) / 1000m;
+
             SecurityIdentifier underlyingSid;
             if (securityType == SecurityType.Option)
             {
                 underlyingSid = SecurityIdentifier.GenerateEquity(optionTicker, market);
-                // let it fallback to it's default handling, which include mapping
+                // Reset optionTicker for default handling
                 optionTicker = null;
             }
-            else if(securityType == SecurityType.IndexOption)
+            else if (securityType == SecurityType.IndexOption)
             {
                 underlyingSid = SecurityIdentifier.GenerateIndex(OptionSymbol.MapToUnderlying(optionTicker, securityType), market);
             }
             else
             {
                 throw new NotImplementedException($"ParseOptionTickerOSI(): {Messages.SymbolRepresentation.SecurityTypeNotImplemented(securityType)}");
             }
+
             var sid = SecurityIdentifier.GenerateOption(expiration, underlyingSid, optionTicker, market, strike, right, optionStyle);
             return new Symbol(sid, ticker, new Symbol(underlyingSid, underlyingSid.Symbol));
         }
+    
+
```

Comment:
Let's also add a new method `GenerateOptionTickerOSICompact` with this new format without spaces and add unit tests for it 👍 

nit - let's revert unrequired space changes like these

Reply:
sure will work on it